app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -70,7 +71,10 @@ protected void onCreate(Bundle savedInstanceState) {
         mSendButton = (Button) findViewById(R.id.sendButton);
         mMessageEdit = (EditText) findViewById(R.id.messageEdit);
 
-        mChatRef = FirebaseDatabase.getInstance().getReference().child("chats");
+        mEmptyListView = findViewById(R.id.emptyTextView);
+
+        mRef = FirebaseDatabase.getInstance().getReference();
+        mChatRef = mRef.child("chats");
```

Comment:
Oh cool! I haven't been able to try that out yet. Github just needs to add checks to make sure your code compiles 😄 (oh, and why not add Intellij while we're at it and just write all of your code for you! 😄)